lib/dialects/abstract/query-generator/select-proto.js

```diff
@@ -0,0 +1,40 @@
+'use strict';
+
+const Composition = require('./composition');
+const QueryProto = require('./query-proto');
+
+class SelectProto extends QueryProto {
+  toComposition() {
+    const composition = new Composition();
+    composition.add('SELECT');
+
+    if (this.attributes.length) {
+      composition.add(' ', this.attributes);
+    } else {
+      composition.add(' *');
+    }
+
+    if (! (this.from && this.from.length)) {
+      throw new Error('Missing FROM clause in select proto query');
```

Comment:
Might be good to have a `CompositionError` class.

Reply:
Done